envisage/ui/tasks/tasks_application.py

```diff
@@ -461,11 +461,8 @@ def _gui_default(self):
         return GUI(splash_screen=self.splash_screen)
 
     def _state_location_default(self):
-        state_location = os.path.join(
-            ETSConfig.application_home, "tasks", ETSConfig.toolkit
-        )
-        if not os.path.exists(state_location):
-            os.makedirs(state_location)
+        state_location = os.path.join(self.home, "tasks", ETSConfig.toolkit)
+        os.makedirs(state_location, mode=0o700, exist_ok=True)
```

Comment:
Possibly should shift this to `_initialize_application_home`?

Reply:
Yes, agreed that extending `_initialize_application_home` seems like the right thing here. I'll make a quick commit to change that.